topics/lua-api.md

```diff
@@ -0,0 +1,745 @@
+
+local function my_local_function()
+  -- Do something else, but equally amazing
+end
+```
+
+**Note:***
+the sandbox attempts to prevent the use of globals.
+Using Lua's debugging functionality or other approaches such as altering the meta table used for implementing the globals' protection to circumvent the sandbox isn't hard.
+However, it is difficult to circumvent the protection by accident.
+If the user messes with the Lua global state, the consistency of AOF and replication can't be guaranteed.
+In other words, just don't do it.
+
+### Imported Lua modules
+
+Using imported Lua modules is not supported inside the sandboxed execution context.
+The sandboxed execution context prevents the loading modules by disabling Lua's [`require` function](https://www.lua.org/pil/8.1.html).
+
+The only libraries that Redis ships with and that you can use in scripts are listed under the [Runtime libraries](#runtime-libraries) section.
+
+## Runtime globals
+
+While the sandbox prevents users from declaring globals, the execution context is pre-populated with several of these.
+
+###cThe _redis_ singleton
+
+The _redis_ singleton is an object instance that's accessible from all scripts.
+It provides the API to interact with Redis from scripts.
+Its description follows [below](#redis_object).
+
+### The _KEYS_ global variable
+
+* Since version: 2.6.0
+* Available in scripts: yes
+* Available in functions: no
+
+**Important:**
+to ensure the correct execution of scripts, both in standalone and clustered deployments, all names of keys that a function accesses must be explicitly provided as input key arguments.
+The script **should only** access keys whose names are given as input arguments.
+Scripts **should never** access keys with programmatically-generated names or based on the contents of data structures stored in the database.
+
+The _KEYS_ global variable is available only for [ephemeral scripts](/topics/eval-intro).
+It is pre-populated with all key name input arguments.
+
+### The _ARGV_ global variable
+
+* Since version: 2.6.0
+* Available in scripts: yes
+* Available in functions: no
+
+The _ARGV_ global variable is available only in [ephemeral scripts](/topics/eval-intro).
+It is pre-populated with all regular input arguments.
+
+## _redis_ object
+
+* Since version: 2.6.0
+* Available in scripts: yes
+* Available in functions: yes
+
+The Redis Lua execution context always provides a singleton instance of an object named _redis_.
+The _redis_ instance enables the script to interact with the Redis server that's running it.
+Following is the API provided by the _redis_ object instance.
+
+### `redis.call(command [,arg...])`
+<a href="redis.call()"></a>
+
+* Since version: 2.6.0
+* Available in scripts: yes
+* Available in functions: yes
+
+The `redis.call()` function calls a given Redis command and returns its reply.
+Its inputs are the command and arguments, and once called, it executes the command in Redis and returns the reply.
```

Comment:
i think this section (speaking about `redis.call` is missing some link to where we talk about OOM errors